app/vmctl/influx/influx.go

```diff
@@ -180,11 +180,7 @@ func (c *Client) Explore() ([]*Series, error) {
 			log.Printf("skip measurement %q since it has no fields", s.Measurement)
 			continue
 		}
-		tags, ok := measurementTags[s.Measurement]
-		if !ok {
-			return nil, fmt.Errorf("failed to find tags of measurement %s", s.Measurement)
-		}
-		emptyTags := getEmptyTags(tags, s.LabelPairs)
+		emptyTags := getEmptyTags(measurementTags[s.Measurement], s.LabelPairs)
```

Comment:
Looks like it's worth to add fast path check for `getEmptyTags` function:

```
if len(tags) == 0 {
  return nil
}
```

Reply:
Also, is it possible that given code panic?

```
func getEmptyTags(tags map[string]struct{}, LabelPairs []LabelPair) []string {
	labelMap := make(map[string]struct{})
	for _, pair := range LabelPairs {
		labelMap[pair.Name] = struct{}{}
	}
         # LabelPairs has some duplicates
         # len(labelMap) = 3
         # len(LabelPairs) = 6
         # result := make([]string, 0, -3)
         result := make([]string, 0, len(labelMap)-len(LabelPairs))
```

Reply:
Wow thanks for the catch. It's a bug calculating the diff of `tags` and `LabelPairs`. Somehow `len(labelMap)` was used.